cmd/mavenExecute.go

```diff
@@ -0,0 +1,110 @@
+package cmd
+
+import (
+	"bytes"
+	"github.com/SAP/jenkins-library/pkg/http"
+	"io"
+	"strings"
+
+	"github.com/SAP/jenkins-library/pkg/command"
+	"github.com/SAP/jenkins-library/pkg/log"
+	"github.com/SAP/jenkins-library/pkg/telemetry"
+)
+
+const mavenExecutable = "mvn"
+
+func mavenExecute(config mavenExecuteOptions, telemetryData *telemetry.CustomData) string {
+	c := command.Command{}
+	output, err := runMavenExecute(&config, &c)
+	if err != nil {
+		log.Entry().WithError(err).Fatal("step execution failed")
+	}
+
+	return output
+}
+
+func runMavenExecute(config *mavenExecuteOptions, command execRunner) (string, error) {
+	stdOutBuf, stdOut := evaluateStdOut(config)
+	command.Stdout(stdOut)
+	command.Stderr(log.Entry().Writer())
+
+	parameters := getParametersFromConfig(config, &http.Client{})
+
+	err := command.RunExecutable(mavenExecutable, parameters...)
+	if err != nil {
+		log.Entry().
+			WithError(err).
+			WithField("command", append([]string{mavenExecutable}, parameters...)).
+			Fatal("failed to execute run command")
+	}
+
+	if stdOutBuf == nil {
+		return "", nil
+	}
+	return string(stdOutBuf.Bytes()), nil
+}
+
+func evaluateStdOut(config *mavenExecuteOptions) (*bytes.Buffer, io.Writer) {
+	var stdOutBuf *bytes.Buffer
+	var stdOut io.Writer
+
+	stdOut = log.Entry().Writer()
+	if config.ReturnStdout {
+		stdOutBuf = new(bytes.Buffer)
+		stdOut = io.MultiWriter(stdOut, stdOutBuf)
+	}
+	return stdOutBuf, stdOut
+}
+
+func getParametersFromConfig(config *mavenExecuteOptions, client http.Downloader) []string {
+	var parameters []string
+
+	if config.GlobalSettingsFile != "" {
```

Comment:
We have the same things to to (... getting maven settings files). What about using shared coding for that (see PR #1176 )?

Reply:
Totally agree and that is the reason I have the comment in line 104. I would propose the PR merged later should just adapt the changes.

Reply:
😄 well, its to some extend a political issue. I would like to keep some pressure in order to get the other PR merged. When that here is merged I suppose nobody is interested in the other PR anymore. And here it is hard to find reviewers.

From the technical point of view I fully agree with your point of view.